ManageIQ/manageiq-ui-classic
# [WIP] Move dialog-editor from ui-components

Part of https://github.com/ManageIQ/manageiq-ui-classic/issues/6716
ui-components PR: https://github.com/ManageIQ/ui-components/pull/436
can be merged in any order

This moves dialog-editor and styles from ui-components to ui-classic,
`src/dialog-editor` is the bulk, `styles/index.scss` is the dialog editor parts of `src/styles/ui-components.scss`, `styles/dialog*editor*` were moved whole from `src/styles`,
and the missing color definitions from `src/styles/colors.scss`.

It also moves our DialogEditorHttp service and the DialogEditorController inside the new `dialog-editor/` code, as `DialogEditorHttp` and a `miq-dialog-editor` component.

Changes to the original:

* converted from typescript to es6,
* linted,
* removed boilerplate comments,
* unified in a flat `whatever.{component,controller,service}.js` + `whatever.html` structure, with the naming matching between files, exports and angular elements (`dialog-editor-toolbox` is `toolbox.component.js` and uses `ToolboxController`, etc.) 
* changed template (html/haml) loading from dynamic requires to use our StaticController to serve the templates from `app/views/static/dialog-editor` - symlinked all the templates in there
* cleaned styles for missing elements (`.delete-tab`, `.new-tab`)

TODO:

* fix tests
* test with&without old copy
* ensure future i18n? (html to haml?)
* lodash cleanup, include https://github.com/ManageIQ/ui-components/pull/429 + https://github.com/ManageIQ/manageiq-ui-classic/pull/6449
* merge miq-dialog-editor with dialog-editor, move shared data to the service instead of copying everywhere
* _.forEach  vs .forEach is different for non-arrays (load errors when editing)
* add changes from https://github.com/ManageIQ/ui-components/pull/437 and later PRs :(
